Flux-User-Guide.md

```diff
@@ -0,0 +1,132 @@
+```bash
+$> flux.sh FILE.flux var1=value1 var2=value2
+```
+This sets the variable `var1` to the value 'value1' and `var2` to the value 'value2'.
+
+## Writing Flux files
+The following snippet shows a simple flux file:
+```c
+//declare variables
+default file = FLUX_DIR + "10.marc21";
+
+//declare flow
+file
+| open-file
+| as-lines
+| decode-marc21
+| fix(FLUX_DIR + "fix-marc21.fix")
+| encode-json(prettyPrinting="true")
+| write("stdout")
+;
+```
+In the first section [variables](#variables) are declared, in the second, we [define the flow](#flow-definitions).
+A flow is a combination of different [FLUX commands. Here is a list to all available Flux-Commands.](https://github.com/metafacture/metafacture-documentation/blob/master/flux-commands.md)
+
+Linebreaks are optional, but help concerning readability. One can add comments with `//`.
+Semicolons `;` mark the end of a variable assignment or flow definition. 
+
+### Flow Definitions
+
+A FLUX contains multiple command-moduls that are doing specific things. E.g.:
+
+```C
+"file/path.mrc"
+| open-file -> This opens the file of the provided `file/path.mrc`-path.
+| as-lines - This reads the file by lines.
+| decode-marc21 -> This decodes the data as binary marc21 into an internal format.
+| fix(FLUX_DIR + "fix-marc21.fix")  -> This executes the provided Fix-transformation.
+| encode-json(prettyPrinting="true") -> This encodes the transformed data as JSON-Format.
+| write("stdout") -> This writes the Json-Data to standard output.
+;
+``````
+
+The syntax for defining flows takes its cues from bash pipes. Commands are concatenated with the pipe character `|`. 
+
+Some commands take a constructor argument. It is provided within brackets: `command("arg")`.
+Furthermore, some commands have named options. These are set as follows `command(optionname="arg1",annotheroption="arg2")` or with constructor argument: `command("arg",option="arg2")`.
+To learn about the available options of a command, execute Flux without arguments: It will list all available commands, including options. or have a look at: [List of available FLUX commands.](https://github.com/metafacture/metafacture-documentation/blob/master/flux-commands.md) TODO: We need to add FIX to that list!!!
+
+
+To some commands, the entire environment can be given as argument. This is done with the `*` character: `fix("tranformation.fix", *)`. In this case Metafix gains access to all variable assignments made in Flux.
+(See also [[Metafix-User-Guide#parameters-to-metafix-definitions]]).
+
+Note that unlike shell pipes, the data flowing between Flux commands is _typed_. This means that only commands with matching signatures can be combined. Commands expect a certain input and provide a certain output like: `StreamReceiver, `Object`, `Reader` and others.
+
+To lookup the signatures, execute Flux without arguments or see: [[Metafix-User-Guide#parameters-to-metafix-definitions]]). It will list all available commands, including signatures.
```

Comment:
```suggestion
To lookup the signatures, again: execute Flux without arguments or see: [[Metafix-User-Guide#parameters-to-metafix-definitions]]). It will list all available commands, including signatures. Or simply have a look at the  [list of available FLUX commands.](https://github.com/metafacture/metafacture-documentation/blob/master/flux-commands.md)
```